ic-cdk/src/call.rs

```diff
@@ -31,41 +28,47 @@ pub enum RejectCode {
     CanisterError = 5,
     /// Response unknown; system stopped waiting for it (e.g., timed out, or system under high load).
     SysUnknown = 6,
+}
 
-    /// Unrecognized reject code.
-    ///
-    /// Note that this variant is not part of the IC interface spec, and is used to represent
-    /// reject codes that are not recognized by the library.
-    Unrecognized(u32),
+/// Error type for [`RejectCode`] conversion.
+///
+/// A reject code is invalid if it is not one of the known reject codes.
+#[derive(Clone, Copy, Debug)]
+pub struct InvalidRejectCode(pub u32);
+
+impl std::fmt::Display for InvalidRejectCode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "invalid reject code: {}", self.0)
+    }
 }
 
-impl From<u32> for RejectCode {
-    fn from(code: u32) -> Self {
+impl Error for InvalidRejectCode {}
+
+impl TryFrom<u32> for RejectCode {
+    type Error = InvalidRejectCode;
+
+    fn try_from(code: u32) -> Result<Self, Self::Error> {
         match code {
-            // 0 is a special code meaning "no error"
-            0 => RejectCode::NoError,
```

Comment:
would it make sense to assert that the code is not zero?

Reply:
There is no need of assertion right? Since there is no variant for 0 now. And the RejectCode instances are constructed using `TryFrom<u32>` which will panic for 0.

Reply:
> And the RejectCode instances are constructed using TryFrom<u32> which will panic for 0.

I missed that panic which is actually dangerous - please see my other comment. Panicking in case of 0 makes sense though since 0 can never become a new reject code.